### PR TITLE
[trcomp] [huggingface_pytorch] [build] Adding support for PyTorch 1.11

### PR DESCRIPTION
**Pull request by codeislife99**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Lokiiiiii** [comment]
@tejaschumbalkar Could you please look into why the tests are blocked ?

**Comment by Lokiiiiii** [comment]
@codeislife99  Please rebase your branch to fix merge conflicts

**Comment by Lokiiiiii** [comment]
This is fast follow up to https://github.com/aws/deep-learning-containers/pull/1824

File: huggingface/pytorch/training/docker/1.11/py3/cu113/Dockerfile.trcomp.gpu [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+# refer to the above page to pull latest Pytorch image
+
+# docker image region us-west-2
+FROM 763104351884.dkr.ecr.us-west-2.amazonaws.com/pytorch-training:1.11.0-gpu-py38-cu113-ubuntu20.04-sagemaker
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# version args
+ARG TRANSFORMERS_VERSION=4.19.2
```
Comment (Lokiiiiii) [inline huggingface/pytorch/training/docker/1.11/py3/cu113/Dockerfile.trcomp.gpu:11]: Let's add a comment here that these arguments are expected to be overwritten by those set in the buildspec. Trying to avoid any confusion for people unfamiliar with the DLC build system.

File: huggingface/pytorch/buildspec-trcomp.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+images:
+  BuildHuggingFacePytorchTrCompGpuPy38Cu113TrainingDockerImage:
+    <<: *TRAINING_REPOSITORY
+    build: &HUGGINGFACE_PYTORCH_TRCOMP_GPU_TRAINING_PY3 false
+    image_size_baseline: &IMAGE_SIZE_BASELINE 18000
+    device_type: &DEVICE_TYPE gpu
+    python_version: &DOCKER_PYTHON_VERSION py3
+    tag_python_version: &TAG_PYTHON_VERSION py38
+    cuda_version: &CUDA_VERSION cu113
```
Comment (Lokiiiiii) [inline huggingface/pytorch/buildspec-trcomp.yml:23]: Base PT just launched [support for cu115](https://github.com/aws/deep-learning-containers/pull/1888). We need to check what [strategy HF is taking](https://github.com/aws/deep-learning-containers/pull/1824) - do we need to support both 113 and 115 ? or are we skipping 113 in favor of 115 ?

File: huggingface/pytorch/training/docker/1.11/py3/cu113/Dockerfile.trcomp.gpu [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+ARG DATASETS_VERSION=1.18.4
+ARG PYTHON=python3
+
+ARG PT_BUCKET=https://sm-training-comp-pytorch-binaries.s3.us-west-2.amazonaws.com/617bcafe-7da3-4700-a4ba-6707a11c6630/20220526-213653/9ad1460a3411c47cc696c97b52d11dd3984fd2b1
```
Comment (Lokiiiiii) [inline huggingface/pytorch/training/docker/1.11/py3/cu113/Dockerfile.trcomp.gpu:15]: [PT 1.11 Base DLC ](https://github.com/aws/deep-learning-containers/blob/master/pytorch/training/docker/1.11/py3/cu115/Dockerfile.gpu#L11) shipped with [TorchData](https://pytorch.org/data/beta/index.html). Please check if we need to rebuild this with trcomp support.

File: huggingface/pytorch/training/docker/1.11/py3/cu113/Dockerfile.trcomp.gpu [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+# COPY ${SAGEMAKER_BINARY} .
+RUN pip install --no-cache-dir sagemaker
+
+
+# RUN pip install --no-cache-dir ${SAGEMAKER_BINARY} \
+#  && rm -rf ${SAGEMAKER_BINARY}
+
+# install Hugging Face libraries and its dependencies
+RUN pip install -U --no-cache-dir \ 
+	transformers[sklearn,sentencepiece]==${TRANSFORMERS_VERSION} \
```
Comment (Lokiiiiii) [inline huggingface/pytorch/training/docker/1.11/py3/cu113/Dockerfile.trcomp.gpu:36]:
```suggestion
	transformers[sklearn,sentencepiece,audio,vision]==${TRANSFORMERS_VERSION} \ 
```

**Comment by codeislife99** [comment]
Made a new PR here #2032
